client/blocks/login/login-form.jsx

```diff
@@ -109,20 +113,42 @@ export class LoginForm extends Component {
 	onSubmitForm = event => {
 		event.preventDefault();
 
-		const { usernameOrEmail, password } = this.state;
-		const { onSuccess, redirectTo } = this.props;
+		const { usernameOrEmail } = this.state;
+		const { accountType } = this.props;
 
-		this.props.recordTracksEvent( 'calypso_login_block_login_form_submit' );
+		if ( isRegularAccount( accountType ) ) {
+			const { password } = this.state;
+			const { onSuccess, redirectTo } = this.props;
 
-		this.props
-			.loginUser( usernameOrEmail, password, redirectTo )
+			this.props.recordTracksEvent( 'calypso_login_block_login_form_submit' );
+
+			this.props
+				.loginUser( usernameOrEmail, password, redirectTo )
+				.then( () => {
+					this.props.recordTracksEvent( 'calypso_login_block_login_form_success' );
+
+					onSuccess( redirectTo );
+				} )
+				.catch( error => {
+					this.props.recordTracksEvent( 'calypso_login_block_login_form_failure', {
+						error_code: error.code,
+						error_message: error.message,
+					} );
+				} );
+
+			return;
+		}
+
+		this.props.recordTracksEvent( 'calypso_login_block_login_form_get_auth_type' );
```

Comment:
I'm not entirely sure how all those new Tracks event should be named, neither if they make sense.